### ISSUE TITLE
Kubelet manifest doesn't allow multiple host ports to map to the same internal port

### ISSUE THREAD

**Issue by JorritSalverda** [kubernetes/kubernetes#13203]
Where the following command

```
docker run -d=true -p 80:8153 -p 443:8154 -p 8154:8154 gocd/gocd-server:15.2.0 /sbin/my_init
```

Results in the following bindings:

```
"PortBindings": {
    "8153/tcp": [
        {
            "HostIp": "",
            "HostPort": "80"
        }
    ],
    "8154/tcp": [
        {
            "HostIp": "",
            "HostPort": "443"
        },
        {
            "HostIp": "",
            "HostPort": "8154"
        }
    ]
}
```
 
Trying to do the same in the kubelet manifest:

```
 ports:
    - name: http
      containerPort: 8153
      hostPort: 80
      protocol: TCP
    - name: https-internal
      containerPort: 8154
      hostPort: 443
      protocol: TCP
    - name: https
      containerPort: 8154
      hostPort: 8154
      protocol: TCP
```

Results in the last port mapping overriding the second one:

```
"PortBindings": {
    "8153/tcp": [
        {
            "HostIp": "",
            "HostPort": "80"
        }
    ],
    "8154/tcp": [
        {
            "HostIp": "",
            "HostPort": "8154"
        }
    ]
}         
```

**Comment by resouer** [comment]
This is interesting, can you tell us some details for your use case?

**Comment by JorritSalverda** [comment]
It's mostly a workaround for go.cd server returning its internal ssl port to the agent on a different port than i would to map ssl to (443 external port, 8154 internal port). See ticket https://github.com/gocd/gocd/issues/1459. In that case having both 443 and 8154 map to 8154 would sort that problem. Outside of this workaround I can't really think of a good reason to use it.

**Comment by resouer** [comment]
i see, the reason is that we using dockerport as the key of portbindings[dockerport]=hostip:port that's why the prior one will be replaced. i can fix this in minutes, if any owner think ur case make sense

so #dibs

**Comment by resouer** [comment]
This can be closed, since #13384 is in @JorritSalverda
